crates/exex/src/serde.rs

```diff
@@ -223,4 +264,108 @@ mod tests {
             panic!("Expected KakarotSerdeError::MultipleIdentifiersFound");
         }
     }
+
+    #[test]
+    fn test_serialize_pointer_not_struct() {
+        // Setup the KakarotSerde instance
+        let mut kakarot_serde = setup_kakarot_serde();
+
+        // Add a new memory segment to the virtual machine (VM).
+        let _ = kakarot_serde.runner.vm.add_memory_segment();
+
+        // Try to serialize pointer with main which is not a struct.
+        if let Err(KakarotSerdeError::IdentifierNotFound { struct_name, expected_type }) =
+            kakarot_serde.serialize_pointers("main", Relocatable { segment_index: 0, offset: 0 })
+        {
+            assert_eq!(struct_name, "main".to_string());
+            assert_eq!(expected_type, Some("struct".to_string()));
+        } else {
+            panic!("Expected KakarotSerdeError::IdentifierNotFound");
+        }
```

Comment:
```suggestion
        panic!("Expected KakarotSerdeError::IdentifierNotFound");
```

why do we need a if/else?

Reply:
Fixed, just used a match to be able to check the validity of the error message